mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1046,7 +1046,43 @@ static Value makeNDMemRef(OpBuilder &b, Value var, uint32_t ndim) {
   return var;
 }
 
-static func::FuncOp createGPUWrapper(ModuleOp module, const KernelIF &kernel) {
+static func::FuncOp createPrefillKernel(const GenParams &params,
+                                        ModuleOp module, Value mem,
+                                        Attribute prefillAttr) {
+  MLIRContext *context = module.getContext();
+  auto loc = module->getLoc();
+  OpBuilder b(context);
+
+  static size_t funcCounter{0};
+  std::string funcName = "prefill_" + std::to_string(funcCounter);
+  ++funcCounter;
```

Comment:
Strange. I don't know how you managed to select these lines because I already removed them a couple of days ago. The current implementation works as you described